stats/api.py

```diff
@@ -0,0 +1,109 @@
+from datetime import datetime, timezone
+from typing import Optional
+
+import requests
+from django.conf import settings
+from ninja import Query, Router
+from ninja.errors import HttpError
+
+from stats.query import QUERY_DATA
+from stats.schemas import Stat, StatInput, StatOutput, StatPeriodicity
+from stats.utils import floor_to_period, parse_date, shift_period
+
+router = Router(tags=["Stats"])
+
+DESCRIPTION = (
+    "Visiteurs orientés : visiteurs ayant consultés une page d'un produit "
+    "ou ayant fait une recherche sur la carte"
+)
+
+
+def fetch_north_star_stats(
+    periodicity: StatPeriodicity,
+    *,
+    since: Optional[int],
+) -> StatOutput:
+
+    def _get_stat_setting(name: str) -> str:
+        value = settings.STATS.get(name)
+        if not value:
+            raise RuntimeError(f"Le paramètre {name} n'est pas configuré.")
+        return value
+
+    def _build_query_payload(
+        periodicity: StatPeriodicity, since: Optional[int]
```

Comment:
```suggestion
        periodicity: StatPeriodicity, since: int | None
```
Cette notation est dépréciée dans les versions de python récente